### PR TITLE
Replacing colon character from attachment file names

### PR DESCRIPTION
**Pull request by pagangil**
- Some test runners on linux were creating result files with ":" (colon character) in the file name. Since this was not a valid filename on all OSes, we throw an error in our API. To fix the number of failures, replacing the colon character with underscore character

### CODE REVIEW

File: src/Agent.Worker/TestResults/TestRunPublisher.cs [hunk 1]
```diff
@@ -289,6 +289,7 @@ private TestAttachmentRequestModel GetAttachmentRequestModel(string attachment)
                 string encodedData = Convert.ToBase64String(bytes);
                 if (encodedData.Length <= TCM_MAX_FILECONTENT_SIZE)
                 {
+                    attachment = attachment.Replace(":", "_");
```
[the remark this answers is not in the document]
Comment (pagangil) [reply src/Agent.Worker/TestResults/TestRunPublisher.cs]: Thanks @navin22 for pointing this out. I've fixed this and doing character replacement just on the file name to be published

File: src/Agent.Worker/TestResults/TestRunPublisher.cs [hunk 2]
```diff
@@ -289,7 +289,10 @@ private TestAttachmentRequestModel GetAttachmentRequestModel(string attachment)
                 string encodedData = Convert.ToBase64String(bytes);
                 if (encodedData.Length <= TCM_MAX_FILECONTENT_SIZE)
                 {
-                    return new TestAttachmentRequestModel(encodedData, Path.GetFileName(attachment), "", GetAttachmentType(attachment));
+                    string attachmentFileName = Path.GetFileName(attachment);
+                    attachmentFileName = attachmentFileName.Replace(":", "_");
```
Comment (navin22) [inline src/Agent.Worker/TestResults/TestRunPublisher.cs]: Please add the comment for why we are doing this only for this character

**Comment by navin22** [comment]
Make sure that the ci passes.

File: src/Agent.Worker/TestResults/TestRunPublisher.cs (hunk 2 above)
Comment (pagangil) [reply src/Agent.Worker/TestResults/TestRunPublisher.cs]: Added

**Comment by pagangil** [comment]
@TingluoHuang Can you please help in merging this PR?

**Pull request merged**